node/libs/roles/src/validator/tests.rs

```diff
@@ -268,3 +276,60 @@ fn test_prepare_qc() {
         assert!(qc.verify(&genesis3).is_err());
     }
 }
+
+#[test]
+fn test_validator_committee_weights() {
+    let ctx = ctx::test_root(&ctx::RealClock);
+    let rng = &mut ctx.rng();
+
+    let setup = Setup::new(rng, 6);
+    // Validators weights
+    let weights = [1000, 600, 800, 6000, 900, 700];
+    // Expected sum of the validators weights
+    let sums = [1000, 1600, 2400, 8400, 9300, 10000];
+    let validators: Vec<WeightedValidator> = weights
+        .iter()
+        .enumerate()
+        .map(|(i, w)| WeightedValidator {
+            key: setup.keys[i].public(),
+            weight: *w,
+        })
+        .collect();
+
+    let genesis = Genesis {
+        validators: ValidatorCommittee::new(validators).unwrap(),
+        fork: setup.genesis.fork.clone(),
+    };
+
+    let view: ViewNumber = rng.gen();
+    let msg = make_replica_prepare(rng, view, &setup);
+    let mut qc = PrepareQC::new(msg.view.clone());
+    for (n, weight) in sums.iter().enumerate() {
+        let key = &setup.keys[n];
+        qc.add(&key.sign_msg(msg.clone()), &setup.genesis);
+        let signers = &qc.map[&msg];
+        assert_eq!(genesis.validators.weight(signers.clone()), *weight);
+    }
+}
+
+#[test]
+fn test_validator_weights_sanity() {
+    let ctx = ctx::test_root(&ctx::RealClock);
+    let rng = &mut ctx.rng();
+
+    let setup = Setup::new(rng, 6);
+    // Validators weights
+    let weight = u64::MAX / 5;
+    let weights = [weight, weight, weight, weight, weight, weight];
```

Comment:
nit: you can equivalently write `[weight;6]`: https://gist.github.com/rust-play/0cc713303e7094544cfc881cb5b65ff6